### PR TITLE
Drop the sinatra dependency

### PR DESCRIPTION
**Pull request by casperisfine**
I was a bit tired of waiting on Sinatra to release a non-beta rack 2 compatible version. 

So I figured let's see how hard it would be to go with only rack. Here's the response.

@gmalette for review please.

### CODE REVIEW

File: lib/pubsubstub/stream.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Stream
+  def initialize(&callback)
+    @callback = callback
+    @closed = false
+  end
+
+  def close
+    return if closed?
```
Comment (gmalette) [inline lib/pubsubstub/stream.rb]: relic of the past, or the return value of `nil/true` is important?

File: lib/pubsubstub/stream.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @closed = true
+  end
+
+  def each(&front)
+    @front = front
+    @callback.call(self)
+    close unless @keep_open
```
Comment (gmalette) [inline lib/pubsubstub/stream.rb]: always close? `@keep_open` isn't defined

Comment (gmalette) [reply lib/pubsubstub/stream.rb]: Is the closing part taken of by itself when trying to write to a disconnected socket, raising an exception and unregistering the subscription? If yes, is `close` even necessary?

**Review by gmalette — COMMENTED** [review]
This is really cool and a very nice improvement.

File: lib/pubsubstub/stream.rb (hunk 2 above)
Comment (casperisfine) [reply lib/pubsubstub/stream.rb]: Yeah my bad. I started from Sinatra's Stream class and removed the things we didn't need. So yeah we always close.

File: lib/pubsubstub/stream.rb (hunk 1 above)
Comment (casperisfine) [reply lib/pubsubstub/stream.rb]: Hum, yeah it kinda is a relic.

**Comment by casperisfine** [comment]
Just fixed both comments.

**Pull request merged**